.gitignore

```diff
@@ -3,6 +3,7 @@
 .classpath
 .settings
 .cache-main
+.idea
```

Comment:
This should go in your [global `.gitignore` file](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore), not here.  So should most of the other entries here, but I made this before learning better practices...